dhcp/pxe.go

```diff
@@ -123,16 +123,13 @@ func SetupPXE(ctx context.Context, rep, req *dhcp4.Packet) bool {
 	return true
 }
 
-func SetFilename(rep *dhcp4.Packet, filename string, nextServer net.IP, pxeClient bool) {
+func SetFilename(rep *dhcp4.Packet, filename string, nextServer net.IP) {
 	file := rep.File()
 	if len(filename) > len(file) {
 		err := errors.New("filename too long, would be truncated")
 		// req CHaddr and XID == req's
 		dhcplog.With("mac", rep.GetCHAddr(), "xid", rep.GetXID(), "filename", filename).Fatal(err)
 	}
-	if pxeClient {
-		rep.SetString(dhcp4.OptionClassID, "PXEClient")
```

Comment:
why is this being removed? im pretty sure we need to respond to pxe clients with this. is it being set elsewhere, maybe?

ref: Table 2-3 and Table 2-5 of http://www.pix.net/software/pxeboot/archive/pxespec.pdf

Reply:
great catch, I was thinking the wrong thing here. Will drop.

Reply:
Actually going from the code I removed it is *not* needed. It was only being set to true when the incoming DHCP request is from *our* iPXE version. `pxeClient` is never set to true in the `!isPacket` branch in the [code](https://github.com/tinkerbell/boots/blob/6f892cb13e7643e9623d278196984d57f31b6e5b/job/dhcp.go#L142-L152). That means that the NIC was happy to boot the tftp file we sent it w/o the said PXEClient ClassID. I still think we should set it anyway though, but I'll move it to `SetupPXE` instead of setting it depending on the iPXE version or what not.

Reply:
`Section 2.2.2: Protocol Timeouts` 2nd paragraph states:
> The purpose of the timeouts is to ensure the PXE client gives precedence to servers supplying
“PXEClient” specific configuration tags. The PXE boot ROM must function as a normal DHCP boot
ROM in the absence of a PXE specific response. However, the PXE boot ROM must wait for
specified times to see if a PXE response is available before using a non-PXE configuration.

So having it is actually not *required* but a very good thing to have anyway.